src/server.js

```diff
@@ -83,41 +94,90 @@ const createServer = async (port, isApi) => {
   await server.register([
     {
       plugin: HapiOpenAPI,
-      options: isApi ? openAPIOptions : openAdminAPIOptions
+      options: openAPIOptions[service]
     }
   ])
   await server.ext([
     {
       type: 'onPreHandler',
       method: (request, h) => {
-        RequestLogger.logResponse(request)
+        // Perform content-type and accept header validation on API routes- not on admin routes,
+        // and not on the health endpoint. Note that hapi-openapi does not correctly validate
+        // headers on routes where the headers are specified at the path level, instead of the
+        // method level. And does not _appear_ to correctly validate the content of `string` +
+        // `pattern` headers at all, although the accuracy of this statement has not been
+        // thoroughly tested.
+        if (service !== serviceType.API || request.path === '/health') {
+          return h.continue
+        }
+
+        // Always validate the accept header for a get request, or optionally if it has been
+        // supplied
+        if (request.method.toLowerCase() === 'get' || request.headers['accept']) {
```

Comment:
This seems like the only feature that needed to be added(i.e this should be in central-services-shared) in this PR and there is no need for the huge refactor. Please advise as to why all the refactoring was done.